lisa/util/subclasses.py

```diff
@@ -47,7 +47,10 @@ def _initialize(self, *args: Any, **kwargs: Any) -> None:
         # initialize types from subclasses.
         # each type should be unique in code, or there is warning message.
         for subclass_type in self._get_subclasses(self._base_type):
-            subclass_type_name = subclass_type.type_name()
+            try:
+                subclass_type_name = subclass_type.type_name()
```

Comment:
When will it happen? If it needs to instant for middle classes, maybe we have to add them. If capture the not implemented error here, it's not useful also.

Reply:
Should I implement type_name() in DeploymentTransformer? We should never have an instance of DeploymentTransformer so I didn't know how to handle this situation.

Reply:
This case occurs when creating a Transformer factory and it tries to get the name of DeploymentTransformer since it is a subclass.

Reply:
Yes, it's confusing. But it looks like the type_name for an abstract class is needed to bypass this error.

Reply:
Okay, I went forward with just adding a class name to the DeploymentTransformer